### PR TITLE
feat(ui,form): add `RadioGroup` & `RadioFormGroup` components

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Feature

#### What is expected?

Add a new `RadioGroup` component in `@ultraviolet/ui` and a new `RadioGroupField` component in `@ultraviolet/form`

## Relevant logs and/or screenshots

Screenshot 2023-08-07 at 14 14 42

### CODE REVIEW

File: packages/ui/src/components/RadioGroup/index.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }),
+    [name, value, onChange, required],
+  )
+
+  return (
+    <RadioGroupContext.Provider value={contextValue}>
+      <Stack gap={1}>
+        <FieldSet className={className}>
+          <Stack gap={1.5}>
+            <Text as="p" variant="bodyStrong">
```
Comment (philibea) [inline packages/ui/src/components/RadioGroup/index.tsx:122]: <Text as="legend"  is not possible ?

File: packages/ui/src/components/RadioGroup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+}: RadioGroupRadioProps) => {
+  const context = useContext(RadioGroupContext)
+
+  if (!context) {
+    throw new Error('RadioGroup.Radio can only be used inside a RadioGroup')
+  }
+
+  const { groupName, onChange, groupValue } = context
+
+  const radioName = `${groupName}-${name ?? ''}`
```
Comment (matthprost) [inline packages/ui/src/components/RadioGroup/index.tsx:51]: If no name is provided, why not generating one?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/RadioGroup/index.tsx (hunk 2 above)
Comment (matthprost) [reply packages/ui/src/components/RadioGroup/index.tsx:51]: By the way can't we make it nested, like `${groupName}.${name ?? ''}` so in the form the value of the radio will be inside of the radio group name

File: packages/ui/src/components/RadioGroup/__stories__/Helper.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Template } from './Template.stories'
+
+export const Helper = Template.bind({})
```
[the remark this answers is not in the document]
Comment (QuiiBz) [reply packages/ui/src/components/RadioGroup/__stories__/Helper.stories.tsx:3]: That would be the same as the `Controlled` story then?

File: packages/ui/src/components/RadioGroup/index.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  className?: string
+  helper?: ReactNode
+  error?: ReactNode
+  direction?: 'row' | 'column'
+  children: ReactNode
+} & Required<Pick<InputHTMLAttributes<HTMLInputElement>, 'onChange' | 'name'>> &
+  Pick<InputHTMLAttributes<HTMLInputElement>, 'required'>
+
+export const RadioGroup = ({
+  label,
```
Comment (matthprost) [inline packages/ui/src/components/RadioGroup/index.tsx:94]: Can we rename it `legend` everywhere? This is not a label there is a difference between label and legend in HTML, we should follow the same naming.

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/RadioGroup/index.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  )
+
+  return (
+    <RadioGroupContext.Provider value={contextValue}>
+      <Stack gap={1}>
+        <FieldSet className={className}>
+          <Stack gap={1.5}>
+            <Text as="p" variant="bodyStrong">
+              <legend>{label}</legend>
+            </Text>
```
Comment (matthprost) [inline packages/ui/src/components/RadioGroup/index.tsx:122]: Let's use `<Text />` as `legend` and add some required icon, I did that for CheckboxGroup 😄 

PS: design was missing that part they will update ZH according to those specs

```suggestion
            <Text as="legend" variant="bodyStrong">
              {label}&nbsp;
              {required ? (
                <StyledRequiredIcon name="asterisk" color="danger" size={8} />
              ) : null}
            </Text>
```

**Review by matthprost — CHANGES_REQUESTED** [review]

File: .changeset/friendly-baboons-poke.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+'@ultraviolet/ui': patch
+---
+
+Add `RadioGroup` component
```
Comment (matthprost) [inline .changeset/friendly-baboons-poke.md:5]: Shouldn't it be a minor ? On form we have minor for the same addition

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/form/src/components/RadioGroupField/__stories__/index.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      },
+    },
+  },
+  title: 'Form/Components/Fields/RadioGroupField',
+} as Meta
+
+export { Playground } from './Playground.stories'
+export { Error } from './Error.stories'
+export { Helper } from './Helper.stories'
+export { Direction } from './Direction.stories'
```
Comment (matthprost) [inline packages/form/src/components/RadioGroupField/__stories__/index.stories.tsx:27]: Can you add a story with required prop and a submit button following what I have done on checkbox: https://github.com/scaleway/ultraviolet/blob/a3aa6381b656cb1a535926b0eb8f839602b94b14/packages/form/src/components/CheckboxGroupField/__stories__/Required.stories.tsx

So we see that when nothing is checked the submit is disabled until a selection is made

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/form/src/components/RadioGroupField/__stories__/index.stories.tsx (hunk 7 above)
Comment (QuiiBz) [reply packages/form/src/components/RadioGroupField/__stories__/index.stories.tsx:27]: Done:

Screenshot 2023-08-29 at 14 26 57

**Review by matthprost — APPROVED** [review]

**Review by PetmePant — APPROVED** [review]

**Pull request merged**